### PR TITLE
Taxi service

### PR DESCRIPTION
**Pull request by EliosPride**

### CODE REVIEW

File: src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    public Car create(Car car) {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement ps = connection.prepareStatement(CREATE_QUERY, Statement.RETURN_GENERATED_KEYS)) {
+            ps.setString(1, car.getModel());
+            ps.setInt(2, car.getYear());
+            ps.executeUpdate();
+            ResultSet resultSet = ps.getGeneratedKeys();
+            if (resultSet.next()) {
+                Long id = resultSet.getLong(1);
+                car.setId(id);
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java]: @EliosPride do it inline

File: src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    @Override
+    public List<Car> getAll() {
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+             Statement statement = connection.createStatement()) {
+            ResultSet resultSet = statement.executeQuery(GET_ALL_QUERY);
+            while (resultSet.next()) {
+                Long id = resultSet.getLong("id");
+                String model = resultSet.getString("model");
+                Integer year = resultSet.getInt("year");
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java]: @EliosPride use parseCar() here

File: src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.model.Car;
+import mate.jdbc.util.ConnectionUtil;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class CarDaoImpl implements CarDao {
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java:15]: @EliosPride forgot annotation Dao

File: src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.model.Driver;
+import mate.jdbc.util.ConnectionUtil;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class DriverDaoImpl implements DriverDao {
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java:15]: @EliosPride forgot annotation Dao

File: src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    public List<Driver> getAll() {
+        List<Driver> drivers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+             Statement statement = connection.createStatement()) {
+            ResultSet resultSet = statement.executeQuery(GET_ALL_QUERY);
+            while (resultSet.next()) {
+                Long id = resultSet.getLong("id");
+                String firstName = resultSet.getString("first_name");
+                String lastName = resultSet.getString("last_name");
+                Long carId = resultSet.getLong("car_id");
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java]: @EliosPride use parseDriver() here

File: src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java (hunk 1 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java]: @Garret-vin +

File: src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java (hunk 2 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java]: @Garret-vin +

File: src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java (hunk 3 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java:15]: @Garret-vin +

File: src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java (hunk 4 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java:15]: @Garret-vin +

File: src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java (hunk 5 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java]: @Garret-vin +

**Pull request merged**